wordpress-mobile/WordPress-iOS
# Stats: Add Subscribers Emails Opens and Clicks Card

Partially implementing #23045
WordPressKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/794

This PR is for implementing Stats Subscribers `Emails Opens and Clicks` Card. 

I'm leaving `Emails Opens and Clicks` Details for another PR.

## Solution

1. Added a `stats/emails/summary` endpoint used on WP.com to fetch Emails Opens and Clicks data https://github.com/wordpress-mobile/WordPressKit-iOS/pull/794
2. Created `StatsSubscribersStore` to follow `Store` -> `VM` -> `VC` pattern that previous tabs used but simplified `StatsSubscribersStore` to use `Combine` instead of `WordPressFlux`, and have as simple interactions as possible. 
3. Updated existing UI to support 2 data columns for Opens and Clicks
4. Connected everything together in `StatsSubscribersViewModel` and added tests

## To test:

1. Enable `Stats Traffic and Subscribers` tab feature flag
2. Open Stats
3. Open Subscribers tab
4. Confirm Emails Opens and Clicks card appear
5. Compare UI with specs IqhXWz3Iir7RMb5XH5gGfZ-fi-882%3A1728
6. Test on sites without any posts, with few posts, and with many posts
7. View More button functionality is out of scope of this PR


| Empty      | With data |
| ----------- | ----------- |
| ![Simulator Screenshot - iPhone 15 - 2024-04-22 at 17 45 13](https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/ba5ce9f5-a2ed-4aed-8549-7329c7af61f5)      | ![Simulator Screenshot - iPhone 15 - 2024-04-22 at 17 44 42](https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/1ec3eef3-1bba-47ca-bc44-98bae12a88b6)       |

![Simulator Screenshot - iPhone 15 - 2024-04-22 at 17 44 51](https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/d256ae32-f129-4a94-bb9c-e47b152422e1)

## Regression Notes
1. Potential unintended areas of impact

Breaking existing 2 column cells (Search Terms, Referrers) whose UI I modified for new requirements

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing

3. What automated tests I added (or what prevented me from doing so)

Unit tests

PR submission checklist:


Testing checklist:
